% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/arnaq.R
\name{arnaq}
\alias{arnaq}
\title{Create an ARNAQ Report}
\usage{
arnaq(
  resources.file = "resources.yml",
  sample.file = "samples.txt",
  model.name = NULL,
  sample.mask = NULL,
  treat.groups = NULL,
  pca.groups = NULL,
  ERCC = FALSE,
  normalise = "VSN",
  svg.export = FALSE,
  max.samples.per.page = 40,
  pca.depth = NULL,
  ERCC.pairs = list(),
  ERCC.combined = TRUE,
  scatter.pairs = list(),
  gene.mask.name = "Genes"
)
}
\arguments{
\item{resources.file}{The file containing project-centric data, including the location of counts
tables and metric tables. The default looks for the file \code{resources.yml} in your working
directory, and is useful when you
have one project per directory.}

\item{model.name}{An optional model name that allows you to have multiple ARNAQ reports in
parallel for the same project. The model name is appended to the project name in ARNAQ outputs.
Note that running ARNAQ will replace its generated data objects in the current R session.}

\item{sample.mask}{A vector of True/False values, with length equal to the number of samples,
indicating which samples to include in the report. This is used to run reports excluding
outliers. \verb{\link{make_outlier_mask}} is a convenient function for generating one. Default
behaviour is to include all samples.}

\item{treat.groups}{A vector of character objects that specifices which of the group columns in
the \code{samples.txt} file are to be considered treatment groups for \code{DESeq2}'s
normalisation. If the group information is not of full rank, this should be a constrained set of
these groups that is of full rank. Default behaviour is to use all groups.}

\item{pca.groups}{A vector of character objects that specify which groups in \code{samples.txt}
to create PCA plots for. Default behaviour
is to only use those groups in \code{treat.groups}.}

\item{ERCC}{A logical value indicating if ERCC spike-ins are included in the gene set.}

\item{normalise}{A character value specifying which normalisation to perform on the data.
\code{VSN} is the default, but this does not run in reaosnable time for large (>=30) numbers
of samples. \code{linear} is the alternative. Any other value will skip normalisation.}

\item{svg.export}{If set to \code{TRUE}, create an additional directory in the output called
\code{svg} with svg versions of every plot created in the report, suitable for inclusion in
papers.}

\item{pca.depth}{How many components to use when preparing PCA plots. Components beyond 2 will
be plotted against the first component, rather than showing every pairwise combination up to the
limit. Default behaviour is to use all components that account for at least 10\% of the variation,
with a minimum of 2.}

\item{ERCC.pairs}{A list of pairs of samples to produce ERCC-specific fold-change plots for.
Each entry in the list should be a length 2 vector of character objects, naming those samples.
Should be an empty list if ERCC spike-ins are not present in the data.}

\item{ERCC.combined}{When \code{TRUE}, the ERCC.pairs plots above will use concentration as the
x-axis, rather than the expected fold-change.}

\item{scatter.pairs}{A list of pairs of samples to produce read counts scatter plots for, to
check consistency between samples that are technical replicates.}

\item{gene.mask.name}{Which set of genes to filter to in preparing the report. This should be a
name that will be present in the \code{gene.masks} list. See the \verb{objects created} vignette for a list
of options. The default of \code{Genes} will use all genes in the reference \code{.gtf} with a non-zero
count for at least one sample, that are not ERCC spike-ins.}

\item{samples.txt}{The file containin sample-centric data. The default looks for the file
\code{samples.txt} in your working directory, and is useful when you have one project per
directory.}

\item{max.sample.per.page}{In projects with a large number of samples, some plots will be
spread across multiple 'pages' in the output report, to keep them readable. Alter the default
value to fine-tune the point at which this behaviour will occur.}
}
\description{
This function creates a report based on the RNA count data, genome resources and metadata
resources, in html format. It uses external files to define specific file locations and
parameters rather than incorporating them all into the function arguments.
}
\details{
\code{resources.yml} contains information that is identical for all samples, or refers to the
project
as a whole.

\code{samples.txt} contains sample-centric information, such as membership of an experimental
group.

Both of these are assumed to remain unchanged through different ARNAQ runs. Arguments to this
function are those options that may change run to run. More details are in ????.

A directory will be created in the current working directory called
\code{QC} to house the output files, over-writing previous versions, if any.

As a side effect of running this function, a number of R objects are created in the session.
The intention is that these are available for downstream analysis, but can also be adjusted
(such as fine-tuning plots). The R console output will also appear in the file \code{arnaq.log}
in your working directory.

ARNAQ will try to optimise by looking in the session for some data before attempting to read
it from the file locations you give it. If the underlying files have changed, use
\verb{\link{arnaq_clear}} to remove ARNAQ's objects before rerunning.
}
\examples{
arnaq()
arnaq("some/other/location/resources.yml", "some/other/location/samples.txt")
arnaq(model.name = "outliers removed", sample.mask = some_mask, normalise = "linear")

}
\seealso{
\verb{\link{make_outlier_mask}}

\verb{\link{arnaq_clear}}
}
