# ARNAQ

ARNAQ stands for *Analysts' RNA QC*. ARNAQ is a tool for exploring RNA sequencing datasets, with an
eye to differential expression or other analysis. It is designed to make reviewing the data and
isolating potential QC issues as simple and swift as possible.

It is not a replacement for the first line QC performed by sequencing platforms and their
associated tools; but instead to supplement them by considering experimental groups and allowing a
more nuanced approach to potentially problematic samples than a simple pass/fail flag.

Analysts who handle a lot of RNA projects will benefit from being able to perform these QC steps
more quickly, and will help avoid copy/paste scripts that are prone to errors where one line is
not updated for the current project. The report contains descriptions of what the plots show,
helping to present this work to collaborators who are less familiar with bioinformatics.

Investigators who have some familiarity with bioinformatics will benefit from shorter, clearer R
scripts where they do not have to delve too deeply into the specifics of the packages used for this
QC analysis, helping them be confident that the analysis they perform is reliable.

In both cases, the `.svg` export options the produces high quality print-ready plots that can be
easily incorporated into papers.

*ARNAQ is named in reference to the board game 'Lost Ruins of Arnak'.*

## Motivation

The goal of ARNAQ is to make the process of producing a context-aware QC report as quick and
error-free as possible, so analyst time can instead be spent elsewhere.

Metadata about the experiment (such as reference genome) and metadata about samples (such as group
membership) is typically available outside R, and as such ARNAQ uses simple files for this
information, which can either be generated by a pipeline or adapted from the files from a pipeline.

With these files created, the process for creating a report can be as simple as a single line:

```{r}
arnaq()
```

The rationale is that the critical parameters for an analysis should not be buried inside a long
list of R calls. By placing them in tables, and permitted automated generation as part of a
pipeline, it is easy to visually check that the correct metadata is being used.

## Installation

Full installation and post-installation instructions are in [this vignette](vignettes/installation.html).
