---
title: "Installing ARNAQ"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Installing ARNAQ}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r opts, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Apart from the R package requirements, ARNAQ also needs [pandoc](https://pandoc.org)
on the system. If you are using Linux, it is simple and convenient to use
[conda](https://anaconda.org/anaconda/conda) to 
create an environment with `pandoc` and the required R packages.

# Installing dependencies with conda

The file `ARNAQ_env.yml` can be used with `conda` to set up a suitable environment. The enviroment
file can be found [here](https://github.com/Queex/arnaq/blob/main/inst/extdata/ARNAQ_env_1.0.yml).
From the command line:

```{bash conda_1, eval=FALSE}
conda env create -f ARNAQ_env_1.0.yml
conda activate arnaq_1.0
```

When you are done, you can use the usual conda command to return to the standard environment.

```{bash conda_2, eval=FALSE}
conda deactivate
```

# Installing dependencies manually

Several of ARNAQ's required packages are not available for Windows through bioconda. For Windows
installation, or if you do not use conda on your system, a manual installation needs to be done.

First install pandoc, from [here](https://pandoc.org/installing.html).

Then make sure the `devtools` package is installed:

```{r pre_install, eval=FALSE}
install.packages("devtools")
```

## Caveats

As of writing this guide, several dependencies of dependencies do not seem to be available for R 
version 4.4 via
this install method. You can try installing them from Bioconductor manually, or use an previous
version of R. If you follow the latter method, one package requires R 4.4 and up, which will raise
a warning, but this does not seem to prevent ARNAQ from running.

# Installing ARNAQ from github

```{r install, eval=FALSE}
devtools::install_github("Queex/arnaq")
```

# Support Files

ARNAQ uses two support files to process the data and create reports, and one file specifying 
metadata about the project. By default, ARNAQ will use internal versions of the first two. For the
third, you will need to create it for every project. You can create this file manually, copy over 
and edit one from a previous project, or create a blank template using the following
command:

```{r prepare, eval=FALSE}
arnaq.prepare("resources")
```

The same function can also be used to create copies of the other two files, if you need to alter
them.

These are stored as files outside R in order to make it possible for them to be automatically
generated (or at least generated in part) by an upstream pipeline.

## resources.yml

This file contains metadata about a project that applies to all samples. The plots that appear in
the ARNAQ report are governed, in part, by which data files are provided in `resources.yml`.

The entries for this file should be:

- **project_id**: this is a name to use for the report, which will also be
prepended to the files the script outputs.
- **species**: a species identifier, which will be listed in the report but not used for any
- processing. This can be a more presentation-ready equivalent of `genome_reference`, next.
- **genome_reference**: the directory name containing the `.gtf` file with gene definitions. See
below for how to set this.
- **count_table**: this is the location of the count table file.
- **summary_table**: this is the location of the read assignment summary,
generated by featureCounts or similar tool. If listed as 'None', it is assumed 
that this data is unavailable and the report will be produced without that 
section.
- **duplication_table**: this is the location of a table of duplication rates
by sample. If listed as 'None', it is assumed to be unavailable.
- **metrics_table**: this is the location of the Picard metrics file. If listed
as 'None', it is assumed to be unavailable.
- **resource_dir**: is the parent directory for all genome references, as described below.
- **report_template**: this is a path to an rmarkdown template used to build the report. The default
of `INTERNAL` will use the standard report in the ARNAQ package. You only need to change this if
you intend to alter the template.
- **biotype_conversion**: the location of a table that maps the full list
of Ensembl biotypes to a smaller number of categories, in order to keep the
associated plots in the report readable. The default of `INTERNAL` will use the standard table
included in the ARNAQ package.
- **ercc_concentrations**: the location of a table of data regarding concentrations of ERCC 
spike-ins in the two different mixes, used for QC plots to test accurate detection
of those spike-ins. This is only needed if you are using ERCC spike-in features of ARNAQ.

# Sample metadata file

This file contains metadata about specific samples in a project.

- **Fastq**: Unused by this tool; but can be used to store the original filenames for each sample,
or some other identifier. If in doubt, this can be left the same as the next column.
- **Name**: identifiers for samples, that correspond to the column names
in the count table. 
- **Display**: names for the samples to display in the QC document. This can be 
identical to the Name column or you can give samples more readable names.
- Any number of additional columns, where each column is one set of groups samples can belong to.
This group information will be used in the report. Group names should not start with a number.
Numerical data will not be handled correctly by ARNAQ; but you can include these columns anyway as
long as you use the `treat.groups` parameter to only include factor-based metadata.

The order of lines in this file does not matter; the **Name** column is used
to assign the names and meta-data to the correct sample. The version of this table stored in the R
session will be sorted to match the columns of the count data.

# External Support Files (.gtf)

*Optional.*

ARNAQ can make use of an external data file, available from public repositories, for each reference
genome you intend to use ARNAQ with. They are not included in ARNAQ's distribution, as they must be
specific for your mapping pipeline and match the gene identifiers in your count files.

Specifically, ARNAQ presumes that Ensembl `.gtf` files for the organism are available. These should
include identifiers drawn from the same set as the gene identifiers uses in the count tables. There
should be a 'type' field, so ARNAQ can identify the gene-level entries. For the biotype portions of
ARNAQ output, there should also be a 'gene_biotype' field. You may wish to filter the Ensembl files
to just the entries with a `type` of `"gene"`, to reduce the size of the file and make ARNAQ runs
faster, but it is not required.

You will only need one such file for each reference genome you are working with.

## Directory Structure

ARNAQ presumes a particular directory structure for these files.

`resource_dir` and `genome_reference` in `resources.yml` define the directory ARNAQ looks in for
the `.gtf` file. The intention is that `resource_dir` gives the directory where a number of named
genomes are placed, and can be kept the same for every project. `genome_reference` names the 
directory in that location where the correct `.gtf` file can be found. ARNAQ will use the first
`.gtf` file it finds in that directory, so you don't have to rename files you download and risk
losing information as to their version.

If your directory looks like this:

```
/some/directory/you/have/
├── GRCh38/
│   └── GRCh38.p14.20240712.gtf
└── GRCm39/
    └── GrCm39.20240317.gtf
```

The matching lines in `resources.yml` would be:

```
resource_dir: /some/directory/you/have/
genome_reference: GRCh38
```

or

```
resource_dir: /some/directory/you/have/
genome_reference: GRCm39
```

The directory path should be an absolute path; so you don't have to adjust it between projects in
different directories.

# ERCC Reference

*Optional.*

If you intend to use the ERCC spike-in functionality of ARNAQ, you will need to download the table
of ERCC ids and mix concentrations from ThermoFisher Scientific. As of this revision, this can be
found at [https://assets.thermofisher.com/TFS-Assets/LSG/manuals/cms_095046.txt](https://assets.thermofisher.com/TFS-Assets/LSG/manuals/cms_095046.txt).

Make sure the relevant line in `resources.yml` points to this file:

```
ercc_concentrations: path/to/file/from/TFS/filename.txt
```