<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Structure of the all_plots list • arnaq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Structure of the all_plots list">
<meta property="og:description" content="arnaq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">arnaq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/all-plots.html">Structure of the all_plots list</a>
    </li>
    <li>
      <a href="../articles/creating-example-reports.html">Creating Example Reports</a>
    </li>
    <li>
      <a href="../articles/files-created.html">Files Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/objects-created.html">Objects Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/workflow.html">Typical ARNAQ Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Structure of the all_plots list</h1>
            
      
      
      <div class="hidden name"><code>all-plots.Rmd</code></div>

    </div>

    
    
<p>It’s only necessary to understand <code>all_plots</code> if you
intend to modify the automatically generated plots.</p>
<p>Before the report is rendered, the plot objects are stored in a list.
This allows the user to adjust any of the plots and re-run
<code><a href="../reference/arnaq.create.report.html">arnaq.create.report()</a></code> or <code><a href="../reference/export_all_svg.html">export_all_svg()</a></code> to
include the updated version in the output. As <code>ggplot2</code>
plots, they can be latered using any of the usual <code>ggplot2</code>
grammar. Alternatively, or in combination with such adjustments, the
functions that create these plots can be used with different parameters.
Plots can be turned by altering the <code>arnaq.plot.flags</code> object
and rerunning one of the above functions.</p>
<p>The <code>_ps</code> suffix is a convention to indicate that the list
entry contains a list of unnamed subplots, usually for clarity where
there are more samples than can comfortably fir on one plot.</p>
<p>The full structure of the object is:</p>
<pre class="text"><code>all_plots
├── $barplots_ps
│   ├── [[1]]
│   ├── [[...]]
│   └── [[i]]
├── $barplots2_ps
│   ├── [[1]]
│   ├── [[...]]
│   └── [[i]]
├── $violin1
├── $violin2
├── $normplot
├── $hcluster_ps
│   ├── [[1]]
│   ├── [[2]]
│   ├── [[3]]
│   └── [[4]]
├── $pca_ids_ps
│   ├── [[1]]
│   ├── [[...]]
│   └── [[i]]
├── $pca_group_pss
│   ├── [[1]]
│   │   ├── [[1]]
│   │   ├── [[...]]
│   │   └── [[i]]
│   ├── [[...]]
│   │   ├── [[1]]
│   │   ├── [[...]]
│   │   └── [[i]]
│   └── [[j]]
│       ├── [[1]]
│       ├── [[...]]
│       └── [[i]]
├── $complexity_plot
├── $complexity_plot_2
├── $detected_plot
├── $detected_plot_2
├── $ercc_mix1_plot
├── $ercc_mix2_plot
├── $ercc_fc_ps
│   ├── [[1]]
│   ├── [[...]]
│   └── [[i]]
├── $biotype_ps
│   ├── [[1]]
│   ├── [[...]]
│   └── [[i]]
├── $biotype_pie_plot
└── $read_scatterplots_ps
    ├── [[1]]
    ├── [[...]]
    └── [[i]]</code></pre>
<div class="section level2">
<h2 id="plot-descriptions">Plot descriptions<a class="anchor" aria-label="anchor" href="#plot-descriptions"></a>
</h2>
<p>Not all plots will be present in every run. Inclusion is based on
factors such as what data is available, how many samples are in the set,
and the parameters supplied to <code><a href="../reference/arnaq.html">arnaq()</a></code>.</p>
<div class="section level3">
<h3 id="barplots_ps-and-barplots2_ps">
<code>barplots_ps</code> and <code>barplots2_ps</code><a class="anchor" aria-label="anchor" href="#barplots_ps-and-barplots2_ps"></a>
</h3>
<p><em>Included when read summary data (as from featureCounts) is given
in <code>resources.yml</code>.</em></p>
<p>These are lists of bar plots showing the read assignment summaries.
In <code>barplots_ps</code>, they are displayed as counts; in
<code>barplots2_ps</code> they are displayed as proportions.</p>
<p>How the samples are split up across multiple plots is controlled by
the <code>max.samples.per.page</code> parameter to <code><a href="../reference/arnaq.html">arnaq()</a></code>.
Note that even when all the samples fit in a single plot, these entries
are still lists of length 1.</p>
</div>
<div class="section level3">
<h3 id="violin1-and-violin2">
<code>violin1</code> and <code>violin2</code><a class="anchor" aria-label="anchor" href="#violin1-and-violin2"></a>
</h3>
<p><em>Included when additional sample metrics (as from Picard) are
given in <code>resources.yml</code>.</em></p>
<p>These show additional metrics in the form of violin plots, split by
group. In this case, only one group is used, rather than every group as
in other plots. The group used is the first named in the
<code>treat.groups</code> parameter.</p>
<p>The metrics are split across two plots, for clarity, based on scale.
Count-based metrics are in <code>violin1</code>. Those based on
proportions are in <code>violin2</code>.</p>
</div>
<div class="section level3">
<h3 id="normplot">
<code>normplot</code><a class="anchor" aria-label="anchor" href="#normplot"></a>
</h3>
<p><em>Included when variance-stabilising normalisation is
performed.</em></p>
<p>This is the standard <code>meanSdPlot</code> in the
<code>DESeq2</code> package.</p>
</div>
<div class="section level3">
<h3 id="hcluster_ps">
<code>hcluster_ps</code><a class="anchor" aria-label="anchor" href="#hcluster_ps"></a>
</h3>
<p><em>Always included.</em></p>
<p>This is a list of four plots showing how the samples cluster. Two
types of plot are show; a sample-to-sample similarity heatmap and a
hiercarchical clustering dendrogram. Two similarity metrics are
considered; Euclidean distance based on the normalised read counts and
Spearman’s rank correlation of the same.</p>
</div>
<div class="section level3">
<h3 id="pca_ids_ps">
<code>pca_ids_ps</code><a class="anchor" aria-label="anchor" href="#pca_ids_ps"></a>
</h3>
<p><em>Always included.</em></p>
<p>These show the PCA plot with samples labelled by their names. Each
requested component from the 2nd onwards is plotted against component 1.
The number of components considered is controlled by
<code>pca.depth</code> and the default is to consider every component
that accounts for 10% of the total variability or more.</p>
<p>Even if there is only a single plot, this is still a list of length
1.</p>
</div>
<div class="section level3">
<h3 id="pca_group_pss">
<code>pca_group_pss</code><a class="anchor" aria-label="anchor" href="#pca_group_pss"></a>
</h3>
<p><em>Always included.</em></p>
<p>This is a list of lists of PCA plots.</p>
<p>Each top level list represents one of the groups specified in
<code>pca.groups</code>. In the case where this is only a single group,
the list will be of length 1.</p>
<p>The sublists have the same format as <code>pca_ids_ps</code>, above,
with one entry per depth of PCA. The samples are coloured by group
membership.</p>
</div>
<div class="section level3">
<h3 id="complexity_plot-and-detected_plot">
<code>complexity_plot</code> and <code>detected_plot</code><a class="anchor" aria-label="anchor" href="#complexity_plot-and-detected_plot"></a>
</h3>
<p><em>Included when the number of samples is &lt;= 40.</em></p>
<p>The complexity plot shows what proportion of the total reads are
assigned to the top N genes, as N varies. The detected genes plot show
how many genes are detected at different thresholds. Each line is one
sample.</p>
<p>In these versions, every sample is shown.</p>
</div>
<div class="section level3">
<h3 id="complexty_plot_2-and-detected_plot_2">
<code>complexty_plot_2</code> and <code>detected_plot_2</code><a class="anchor" aria-label="anchor" href="#complexty_plot_2-and-detected_plot_2"></a>
</h3>
<p><em>Included when the number of samples is &gt; 20.</em></p>
<p>As above, but in this case only the worst 8 samples for each plot are
shown in colour. The remaining samples in the set are coloured grey,
behind.</p>
</div>
<div class="section level3">
<h3 id="ercc_mix1_plot-and-ercc_mix2_plot">
<code>ercc_mix1_plot</code> and <code>ercc_mix2_plot</code><a class="anchor" aria-label="anchor" href="#ercc_mix1_plot-and-ercc_mix2_plot"></a>
</h3>
<p><em>Included when <code>ERCC</code> is <code>TRUE</code> and the
there is an <code>ERCC</code> column in the sample metadata.</em></p>
<p>These two plots show the observed cpm for ERCC spike-in mixes against
the expected.</p>
</div>
<div class="section level3">
<h3 id="ercc_fc_ps">
<code>ercc_fc_ps</code><a class="anchor" aria-label="anchor" href="#ercc_fc_ps"></a>
</h3>
<p><em>Included when <code>ERCC</code> is <code>TRUE</code> and at least
one pair of samples has been given in <code>ERCC.pairs</code>.</em></p>
<p>A list of plots showing observed ERCC spike-in fold-changes between
two samples with different mixes against the known fold-changes.</p>
</div>
<div class="section level3">
<h3 id="biotype_ps">
<code>biotype_ps</code><a class="anchor" aria-label="anchor" href="#biotype_ps"></a>
</h3>
<p><em>Included always.</em></p>
<p>A list of barplots showing the counts of biotypes for the observed
reads, as given in the <code>.gtf</code> support file and mapped to a
smaller nub mer of categories by the biotype conversion table. Where the
number of samples exceeds <code>max.samples.per.page</code>, this is
split across multiple plots for clarity, as with
<code>barplots_ps</code>. Similarly, if all the samples fit on one plot,
the list will be of length 1.</p>
</div>
<div class="section level3">
<h3 id="biotype_pie">
<code>biotype_pie</code><a class="anchor" aria-label="anchor" href="#biotype_pie"></a>
</h3>
<p><em>Included always.</em></p>
<p>A pie chart showing the biotypes of observed genes, not with respect
to read abundance.</p>
</div>
<div class="section level3">
<h3 id="read_scatterplots_ps">
<code>read_scatterplots_ps</code><a class="anchor" aria-label="anchor" href="#read_scatterplots_ps"></a>
</h3>
<p><em>Included when the length of <code>scatter.pairs</code> is at
least 1.</em></p>
<p>A list of plots showing sample-to-sample comparisons, one point per
gene.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by .</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
