<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Creating Example Reports • arnaq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Creating Example Reports">
<meta property="og:description" content="arnaq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">arnaq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/all-plots.html">Structure of the all_plots list</a>
    </li>
    <li>
      <a href="../articles/creating-example-reports.html">Creating Example Reports</a>
    </li>
    <li>
      <a href="../articles/files-created.html">Files Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/objects-created.html">Objects Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/workflow.html">Typical ARNAQ Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Creating Example Reports</h1>
            
      
      
      <div class="hidden name"><code>creating-example-reports.Rmd</code></div>

    </div>

    
    
<p>Example data is provided with ARNAQ; bulk RNA sequencing data from
GEO under the accession <em>GSE155423</em>. This vignette shows the
simplest form of report, with the smallest set of plots, based on this
data set.</p>
<div class="section level2">
<h2 id="initial-arnaq-run">Initial ARNAQ run<a class="anchor" aria-label="anchor" href="#initial-arnaq-run"></a>
</h2>
<div class="section level3">
<h3 id="setting-up-the-relevant-files">Setting up the relevant files<a class="anchor" aria-label="anchor" href="#setting-up-the-relevant-files"></a>
</h3>
<p>These files would normally be prepared by hand, or as part of your
pipeline, but are created here using R for demonstration purposes.</p>
<div class="section level4">
<h4 id="creating-resources-yml">Creating <code>resources.yml</code><a class="anchor" aria-label="anchor" href="#creating-resources-yml"></a>
</h4>
<p>Metadata for the project is stored in a file,
<code>resource.yml</code>. We begin by creating an empty version of this
file using ARNAQ tools.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.prepare.html">arnaq.prepare</a></span><span class="op">(</span><span class="st">"resources"</span><span class="op">)</span> <span class="co"># Creates a template in the working directory</span></span>
<span><span class="co">#&gt; Creating resources.yml in working directory.</span></span>
<span><span class="va">tmp_resources</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"resources.yml"</span>, header<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">tmp_resources</span>  <span class="co"># Shows us what the template looks like</span></span>
<span><span class="co">#&gt;                      V1                                  V2</span></span>
<span><span class="co">#&gt; 1           project_id:                                ????</span></span>
<span><span class="co">#&gt; 2              species:                                ????</span></span>
<span><span class="co">#&gt; 3     genome_reference:                                None</span></span>
<span><span class="co">#&gt; 4          count_table:                                ????</span></span>
<span><span class="co">#&gt; 5        summary_table:                                None</span></span>
<span><span class="co">#&gt; 6    duplication_table:                                None</span></span>
<span><span class="co">#&gt; 7        metrics_table:                                None</span></span>
<span><span class="co">#&gt; 8         resource_dir: /path/to/your/structured/gtf/files/</span></span>
<span><span class="co">#&gt; 9      report_template:                            INTERNAL</span></span>
<span><span class="co">#&gt; 10  biotype_conversion:                            INTERNAL</span></span>
<span><span class="co">#&gt; 11 ercc_concentrations:                                None</span></span></code></pre></div>
<p>This creates a data frame we can alter to point it to the example
data. The minimal version only requires us to add 3 elements.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE155423_counts.txt"</span>, package<span class="op">=</span><span class="st">"arnaq"</span><span class="op">)</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"GSE155423"</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"mouse"</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">4</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">count.file</span></span>
<span><span class="va">tmp_resources</span> <span class="co"># Your count_table location will most likely differ</span></span>
<span><span class="co">#&gt;                      V1</span></span>
<span><span class="co">#&gt; 1           project_id:</span></span>
<span><span class="co">#&gt; 2              species:</span></span>
<span><span class="co">#&gt; 3     genome_reference:</span></span>
<span><span class="co">#&gt; 4          count_table:</span></span>
<span><span class="co">#&gt; 5        summary_table:</span></span>
<span><span class="co">#&gt; 6    duplication_table:</span></span>
<span><span class="co">#&gt; 7        metrics_table:</span></span>
<span><span class="co">#&gt; 8         resource_dir:</span></span>
<span><span class="co">#&gt; 9      report_template:</span></span>
<span><span class="co">#&gt; 10  biotype_conversion:</span></span>
<span><span class="co">#&gt; 11 ercc_concentrations:</span></span>
<span><span class="co">#&gt;                                                                                         V2</span></span>
<span><span class="co">#&gt; 1                                                                                GSE155423</span></span>
<span><span class="co">#&gt; 2                                                                                    mouse</span></span>
<span><span class="co">#&gt; 3                                                                                     None</span></span>
<span><span class="co">#&gt; 4  /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_counts.txt</span></span>
<span><span class="co">#&gt; 5                                                                                     None</span></span>
<span><span class="co">#&gt; 6                                                                                     None</span></span>
<span><span class="co">#&gt; 7                                                                                     None</span></span>
<span><span class="co">#&gt; 8                                                      /path/to/your/structured/gtf/files/</span></span>
<span><span class="co">#&gt; 9                                                                                 INTERNAL</span></span>
<span><span class="co">#&gt; 10                                                                                INTERNAL</span></span>
<span><span class="co">#&gt; 11                                                                                    None</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">tmp_resources</span>, <span class="st">"resources.yml"</span>, quote<span class="op">=</span><span class="cn">FALSE</span>, sep<span class="op">=</span><span class="st">" "</span>, col.names<span class="op">=</span><span class="cn">FALSE</span>, row.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Note that while we’re using R to edit this file for clarity and
reproducibility, it will usually be more convenient to edit these fields
in a text editor.</p>
</div>
<div class="section level4">
<h4 id="creating-samples-txt">Creating <code>samples.txt</code><a class="anchor" aria-label="anchor" href="#creating-samples-txt"></a>
</h4>
<p>The sample metadata will usually come from your own records or
pipeline. ARNAQ includes a pre-built file for this data set.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE155423_samples.txt"</span>, package<span class="op">=</span><span class="st">"arnaq"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="va">samples.file</span>, <span class="st">"./samples.txt"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"./samples.txt"</span>, header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;         Fastq       Name             Display Group Day</span></span>
<span><span class="co">#&gt; 1  SIN6175A19 SIN6175A19 D21Myo_1_but_longer   MYO D21</span></span>
<span><span class="co">#&gt; 2  SIN6175A20 SIN6175A20            D21Myo_2   MYO D21</span></span>
<span><span class="co">#&gt; 3  SIN6175A23 SIN6175A23            D21Myo_3   MYO D21</span></span>
<span><span class="co">#&gt; 4  SIN6175A22 SIN6175A22            D21Myo_4   MYO D21</span></span>
<span><span class="co">#&gt; 5   SIN6542A1  SIN6542A1            D10PBS_1  CTRL D10</span></span>
<span><span class="co">#&gt; 6   SIN6542A2  SIN6542A2            D10PBS_2  CTRL D10</span></span>
<span><span class="co">#&gt; 7   SIN6542A3  SIN6542A3            D10PBS_3  CTRL D10</span></span>
<span><span class="co">#&gt; 8   SIN6542A4  SIN6542A4            D10PBS_4  CTRL D10</span></span>
<span><span class="co">#&gt; 9   SIN6542A5  SIN6542A5            D10Myo_1   MYO D10</span></span>
<span><span class="co">#&gt; 10  SIN6542A6  SIN6542A6            D10Myo_2   MYO D10</span></span>
<span><span class="co">#&gt; 11  SIN6542A7  SIN6542A7            D10Myo_3   MYO D10</span></span>
<span><span class="co">#&gt; 12  SIN6542A8  SIN6542A8            D10Myo_4   MYO D10</span></span>
<span><span class="co">#&gt; 13  SIN6542A9  SIN6542A9            D15PBS_1  CTRL D15</span></span>
<span><span class="co">#&gt; 14 SIN6542A10 SIN6542A10            D15PBS_2  CTRL D15</span></span>
<span><span class="co">#&gt; 15 SIN6542A11 SIN6542A11            D15PBS_3  CTRL D15</span></span>
<span><span class="co">#&gt; 16 SIN6542A12 SIN6542A12            D15PBS_4  CTRL D15</span></span>
<span><span class="co">#&gt; 17 SIN6542A13 SIN6542A13            D15Myo_1   MYO D15</span></span>
<span><span class="co">#&gt; 18 SIN6542A14 SIN6542A14            D15Myo_2   MYO D15</span></span>
<span><span class="co">#&gt; 19 SIN6542A15 SIN6542A15            D15Myo_3   MYO D15</span></span>
<span><span class="co">#&gt; 20 SIN6542A16 SIN6542A16            D15Myo_4   MYO D15</span></span>
<span><span class="co">#&gt; 21 SIN6542A17 SIN6542A17            D21PBS_1  CTRL D21</span></span>
<span><span class="co">#&gt; 22 SIN6542A18 SIN6542A18            D21PBS_2  CTRL D21</span></span>
<span><span class="co">#&gt; 23 SIN6542A19 SIN6542A19            D21PBS_3  CTRL D21</span></span>
<span><span class="co">#&gt; 24 SIN6542A20 SIN6542A20            D21PBS_4  CTRL D21</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="running-the-report">Running the report<a class="anchor" aria-label="anchor" href="#running-the-report"></a>
</h3>
<p>Creating the first report is as simple as:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; tlmgr option sys_bin ~/bin</span></span>
<span><span class="co">#&gt; You may have to restart your system after installing TinyTeX to make sure ~/bin appears in your PATH variable (https://github.com/rstudio/tinytex/issues/16).</span></span>
<span><span class="co">#&gt; processing file: GSE155423_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: GSE155423_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_arnaq.html</span></span>
<span><span class="co">#&gt; processing file: GSE155423_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: GSE155423_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_arnaq.pdf</span></span></code></pre></div>
<p>This will create a new directory in your working directory
<code>QC</code>, which contains not only <code>.pdf</code> and
<code>.html</code> versions of the report but saved count tables and
other processed data.</p>
<p>Additionally, it has also created a number of objects in R, ones that
can be used to refine the QC further or be used for downstream
analysis.</p>
<p>This report is a minimal one, without any plots based on metrics not
supplied. We can fix that.</p>
</div>
</div>
<div class="section level2">
<h2 id="second-run-with-metrics">Second run with metrics<a class="anchor" aria-label="anchor" href="#second-run-with-metrics"></a>
</h2>
<div class="section level3">
<h3 id="additional-metrics">Additional metrics<a class="anchor" aria-label="anchor" href="#additional-metrics"></a>
</h3>
<p>The first step is get the additional metrics files.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assignment_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE155423_assignment.txt"</span>, package<span class="op">=</span><span class="st">"arnaq"</span><span class="op">)</span></span>
<span><span class="va">metrics_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE155423_metrics.txt"</span>, package<span class="op">=</span><span class="st">"arnaq"</span><span class="op">)</span></span>
<span><span class="va">dup_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE155423_dup_metrics.txt"</span>, package<span class="op">=</span><span class="st">"arnaq"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">assignment_file</span>, header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                SIN6542A1 SIN6542A2 SIN6542A3 SIN6542A4 SIN6542A5 SIN6542A6</span></span>
<span><span class="co">#&gt; Assigned        27770637  32072796  29601815  27593406  31453330  28847835</span></span>
<span><span class="co">#&gt; Ambiguity         661332    761268    705771    661909    848845    732008</span></span>
<span><span class="co">#&gt; MultiMapping    12470735  14056355  12985425  10418213  11314804  12350749</span></span>
<span><span class="co">#&gt; NoFeatures       1275619   1469813   1252438   1052594   1557830   1155476</span></span>
<span><span class="co">#&gt; Unmapped          418297    443580    412845    382028    519746    373920</span></span>
<span><span class="co">#&gt; MappingQuality         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt;                SIN6542A7 SIN6542A8 SIN6542A9 SIN6542A10 SIN6542A11 SIN6542A12</span></span>
<span><span class="co">#&gt; Assigned        28528135  30727055  30841433   27794263   32822298   27602676</span></span>
<span><span class="co">#&gt; Ambiguity         718407    786971    741261     689052     846293     637780</span></span>
<span><span class="co">#&gt; MultiMapping    12565193  12249501  14338525   13005171   11398189   11626356</span></span>
<span><span class="co">#&gt; NoFeatures       1170686   1535329   1268341    1212515    1365047    1170523</span></span>
<span><span class="co">#&gt; Unmapped          398969    489411    382949     350670     406881     378272</span></span>
<span><span class="co">#&gt; MappingQuality         0         0         0          0          0          0</span></span>
<span><span class="co">#&gt;                SIN6542A13 SIN6542A14 SIN6542A15 SIN6542A16 SIN6542A17</span></span>
<span><span class="co">#&gt; Assigned         28443782   32712973   30404023   30098105   30183568</span></span>
<span><span class="co">#&gt; Ambiguity          776194     887609     851567     824423     760169</span></span>
<span><span class="co">#&gt; MultiMapping      5896417    4477141    8169773    6264516   12027087</span></span>
<span><span class="co">#&gt; NoFeatures        1547060    1870891    1645683    1886373    1364372</span></span>
<span><span class="co">#&gt; Unmapped           534838     680234     606303     627920     441080</span></span>
<span><span class="co">#&gt; MappingQuality          0          0          0          0          0</span></span>
<span><span class="co">#&gt;                SIN6542A18 SIN6542A19 SIN6542A20 SIN6175A19 SIN6175A20</span></span>
<span><span class="co">#&gt; Assigned         34486901   36798390   29651055   25534610   26096748</span></span>
<span><span class="co">#&gt; Ambiguity          884420     921072     724139     656777     638011</span></span>
<span><span class="co">#&gt; MultiMapping     15289325   15800658   11737910    6719551    9075166</span></span>
<span><span class="co">#&gt; NoFeatures        1321073    1877504    1259688    1006697     919543</span></span>
<span><span class="co">#&gt; Unmapped           429975     487545     365707     457439     402750</span></span>
<span><span class="co">#&gt; MappingQuality          0          0          0          0          0</span></span>
<span><span class="co">#&gt;                SIN6175A23 SIN6175A22</span></span>
<span><span class="co">#&gt; Assigned         23133463   22279175</span></span>
<span><span class="co">#&gt; Ambiguity          569545     587016</span></span>
<span><span class="co">#&gt; MultiMapping      7989749    5389886</span></span>
<span><span class="co">#&gt; NoFeatures        1419573    1016029</span></span>
<span><span class="co">#&gt; Unmapped           561689     438311</span></span>
<span><span class="co">#&gt; MappingQuality          0          0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">metrics_file</span>, header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;      Sample   PF_BASES PF_ALIGNED_BASES RIBOSOMAL_BASES CODING_BASES  UTR_BASES</span></span>
<span><span class="co">#&gt; 1 SIN6542A1 5421859159       5232768652              NA   3351121164 1629190784</span></span>
<span><span class="co">#&gt; 2 SIN6542A2 6225617083       6019587980              NA   3907328151 1820971043</span></span>
<span><span class="co">#&gt; 3 SIN6542A3 5730729023       5541190679              NA   3604348789 1684251923</span></span>
<span><span class="co">#&gt; 4 SIN6542A4 5202591943       5032382031              NA   3370075709 1450681170</span></span>
<span><span class="co">#&gt; 5 SIN6542A5 5959269999       5735116718              NA   3572455394 1854549958</span></span>
<span><span class="co">#&gt; 6 SIN6542A6 5556776115       5365405087              NA   3416601815 1715925386</span></span>
<span><span class="co">#&gt;   INTRONIC_BASES INTERGENIC_BASES IGNORED_READS CORRECT_STRAND_READS</span></span>
<span><span class="co">#&gt; 1      108551189        143905515             0                    0</span></span>
<span><span class="co">#&gt; 2      134004748        157284038             0                    0</span></span>
<span><span class="co">#&gt; 3      107905412        144684555             0                    0</span></span>
<span><span class="co">#&gt; 4       90910449        120714703             0                    0</span></span>
<span><span class="co">#&gt; 5      131550478        176560888             0                    0</span></span>
<span><span class="co">#&gt; 6       94844959        138032927             0                    0</span></span>
<span><span class="co">#&gt;   INCORRECT_STRAND_READS PCT_RIBOSOMAL_BASES PCT_CODING_BASES PCT_UTR_BASES</span></span>
<span><span class="co">#&gt; 1                      0                  NA         0.640411      0.311344</span></span>
<span><span class="co">#&gt; 2                      0                  NA         0.649102      0.302508</span></span>
<span><span class="co">#&gt; 3                      0                  NA         0.650465      0.303951</span></span>
<span><span class="co">#&gt; 4                      0                  NA         0.669678      0.288269</span></span>
<span><span class="co">#&gt; 5                      0                  NA         0.622909      0.323367</span></span>
<span><span class="co">#&gt; 6                      0                  NA         0.636784      0.319813</span></span>
<span><span class="co">#&gt;   PCT_INTRONIC_BASES PCT_INTERGENIC_BASES PCT_MRNA_BASES PCT_USABLE_BASES</span></span>
<span><span class="co">#&gt; 1           0.020745             0.027501       0.951755         0.918562</span></span>
<span><span class="co">#&gt; 2           0.022261             0.026129       0.951610         0.920117</span></span>
<span><span class="co">#&gt; 3           0.019473             0.026111       0.954416         0.922850</span></span>
<span><span class="co">#&gt; 4           0.018065             0.023988       0.957947         0.926607</span></span>
<span><span class="co">#&gt; 5           0.022938             0.030786       0.946276         0.910683</span></span>
<span><span class="co">#&gt; 6           0.017677             0.025726       0.956596         0.923652</span></span>
<span><span class="co">#&gt;   PCT_CORRECT_STRAND_READS MEDIAN_CV_COVERAGE MEDIAN_5PRIME_BIAS</span></span>
<span><span class="co">#&gt; 1                        0           0.406324           0.472710</span></span>
<span><span class="co">#&gt; 2                        0           0.394196           0.509954</span></span>
<span><span class="co">#&gt; 3                        0           0.398620           0.523606</span></span>
<span><span class="co">#&gt; 4                        0           0.397296           0.535933</span></span>
<span><span class="co">#&gt; 5                        0           0.413814           0.483539</span></span>
<span><span class="co">#&gt; 6                        0           0.409612           0.454800</span></span>
<span><span class="co">#&gt;   MEDIAN_3PRIME_BIAS MEDIAN_5PRIME_TO_3PRIME_BIAS SAMPLE LIBRARY READ_GROUP</span></span>
<span><span class="co">#&gt; 1           0.449082                     0.883518     NA      NA         NA</span></span>
<span><span class="co">#&gt; 2           0.434814                     0.943924     NA      NA         NA</span></span>
<span><span class="co">#&gt; 3           0.434651                     0.953354     NA      NA         NA</span></span>
<span><span class="co">#&gt; 4           0.435428                     1.006719     NA      NA         NA</span></span>
<span><span class="co">#&gt; 5           0.449169                     0.906718     NA      NA         NA</span></span>
<span><span class="co">#&gt; 6           0.459661                     0.825427     NA      NA         NA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">dup_file</span>, header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;      Sample PERCENT_DUPLICATION</span></span>
<span><span class="co">#&gt; 1 SIN6542A1            0.406670</span></span>
<span><span class="co">#&gt; 2 SIN6542A2            0.418684</span></span>
<span><span class="co">#&gt; 3 SIN6542A3            0.411908</span></span>
<span><span class="co">#&gt; 4 SIN6542A4            0.436252</span></span>
<span><span class="co">#&gt; 5 SIN6542A5            0.347577</span></span>
<span><span class="co">#&gt; 6 SIN6542A6            0.400317</span></span></code></pre></div>
<p>The first file here is output as from featureCounts, showing how the
reads were assigned to categories.</p>
<p>The other two are Picard output from the modes
<code>CollectRnaSeqMetrics</code> and <code>MarkDuplicates</code>.</p>
<p>You do not need to specify all three; however if you have one of the
Picard outputs ARNAQ expects the other.</p>
<p>Equivalent outputs from other tools should also work.</p>
<p>The names of these files now need to be added to the
<code>resources.yml</code> file.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">assignment_file</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">6</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">dup_file</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">7</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">metrics_file</span></span>
<span><span class="va">tmp_resources</span> <span class="co"># Your locations will most likely differ</span></span>
<span><span class="co">#&gt;                      V1</span></span>
<span><span class="co">#&gt; 1           project_id:</span></span>
<span><span class="co">#&gt; 2              species:</span></span>
<span><span class="co">#&gt; 3     genome_reference:</span></span>
<span><span class="co">#&gt; 4          count_table:</span></span>
<span><span class="co">#&gt; 5        summary_table:</span></span>
<span><span class="co">#&gt; 6    duplication_table:</span></span>
<span><span class="co">#&gt; 7        metrics_table:</span></span>
<span><span class="co">#&gt; 8         resource_dir:</span></span>
<span><span class="co">#&gt; 9      report_template:</span></span>
<span><span class="co">#&gt; 10  biotype_conversion:</span></span>
<span><span class="co">#&gt; 11 ercc_concentrations:</span></span>
<span><span class="co">#&gt;                                                                                              V2</span></span>
<span><span class="co">#&gt; 1                                                                                     GSE155423</span></span>
<span><span class="co">#&gt; 2                                                                                         mouse</span></span>
<span><span class="co">#&gt; 3                                                                                          None</span></span>
<span><span class="co">#&gt; 4       /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_counts.txt</span></span>
<span><span class="co">#&gt; 5   /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_assignment.txt</span></span>
<span><span class="co">#&gt; 6  /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_dup_metrics.txt</span></span>
<span><span class="co">#&gt; 7      /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_metrics.txt</span></span>
<span><span class="co">#&gt; 8                                                           /path/to/your/structured/gtf/files/</span></span>
<span><span class="co">#&gt; 9                                                                                      INTERNAL</span></span>
<span><span class="co">#&gt; 10                                                                                     INTERNAL</span></span>
<span><span class="co">#&gt; 11                                                                                         None</span></span></code></pre></div>
<p>As before, in normal usage you will edit these details outside R.
Also, you will probably only need to set up the
<code>resources.yml</code> file once for each project; rather than
adding lines piecemeal as is done here for demonstration.</p>
</div>
<div class="section level3">
<h3 id="reference--gtf">Reference .gtf<a class="anchor" aria-label="anchor" href="#reference--gtf"></a>
</h3>
<p>Part of ARNAQ’s report uses biotype information from Ensembl gene set
data. These are specific to both the organism and the build of the
genome uses for mapping, so there will likely be a need to keep several
such <code>.gtf</code> files in parallel. ARNAQ assumes a specific
directory structure for where these references are stored, to try to
minimise the time spent setting up each individual project.</p>
<p>In this example we will put this structure in the working directory;
although it is better to put it in a more general location for regular
usage.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="st">"gtfs"</span>, showWarnings<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="st">"gtfs/GRCm39.112"</span>, showWarnings<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://ftp.ensembl.org/pub/release-112/gtf/mus_musculus/Mus_musculus.GRCm39.112.gtf.gz"</span>, <span class="st">"gtfs/GRCm39.112/Mus_musculus.GRCm39.112.gtf.gz"</span><span class="op">)</span></span>
<span><span class="fu">R.utils</span><span class="fu">::</span><span class="fu"><a href="https://henrikbengtsson.github.io/R.utils/reference/compressFile.html" class="external-link">gunzip</a></span><span class="op">(</span><span class="st">"gtfs/GRCm39.112/Mus_musculus.GRCm39.112.gtf.gz"</span><span class="op">)</span></span></code></pre></div>
<p><em>Note that ARNAQ only requires the lines in the <code>.gtf</code>
file for genes; you can filter these files to reduce their size and
speed up ARNAQ’s execution.</em></p>
<p>Now we add the relevant lines to <code>resources.yml</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">8</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"gtfs/"</span> <span class="co"># The top directory where the gene sets are stored</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">3</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"GRCm39.112"</span> <span class="co"># The name of the directory containing our desired .gtf file</span></span>
<span><span class="va">tmp_resources</span><span class="op">[</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"GRCm39.112"</span> <span class="co"># We can be more specific now</span></span>
<span><span class="va">tmp_resources</span> <span class="co"># Your locations will most likely differ</span></span>
<span><span class="co">#&gt;                      V1</span></span>
<span><span class="co">#&gt; 1           project_id:</span></span>
<span><span class="co">#&gt; 2              species:</span></span>
<span><span class="co">#&gt; 3     genome_reference:</span></span>
<span><span class="co">#&gt; 4          count_table:</span></span>
<span><span class="co">#&gt; 5        summary_table:</span></span>
<span><span class="co">#&gt; 6    duplication_table:</span></span>
<span><span class="co">#&gt; 7        metrics_table:</span></span>
<span><span class="co">#&gt; 8         resource_dir:</span></span>
<span><span class="co">#&gt; 9      report_template:</span></span>
<span><span class="co">#&gt; 10  biotype_conversion:</span></span>
<span><span class="co">#&gt; 11 ercc_concentrations:</span></span>
<span><span class="co">#&gt;                                                                                              V2</span></span>
<span><span class="co">#&gt; 1                                                                                     GSE155423</span></span>
<span><span class="co">#&gt; 2                                                                                    GRCm39.112</span></span>
<span><span class="co">#&gt; 3                                                                                    GRCm39.112</span></span>
<span><span class="co">#&gt; 4       /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_counts.txt</span></span>
<span><span class="co">#&gt; 5   /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_assignment.txt</span></span>
<span><span class="co">#&gt; 6  /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_dup_metrics.txt</span></span>
<span><span class="co">#&gt; 7      /home/runner/miniconda3/envs/arnaq_1.0/lib/R/library/arnaq/extdata/GSE155423_metrics.txt</span></span>
<span><span class="co">#&gt; 8                                                                                         gtfs/</span></span>
<span><span class="co">#&gt; 9                                                                                      INTERNAL</span></span>
<span><span class="co">#&gt; 10                                                                                     INTERNAL</span></span>
<span><span class="co">#&gt; 11                                                                                         None</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">tmp_resources</span>, <span class="st">"resources.yml"</span>, quote<span class="op">=</span><span class="cn">FALSE</span>, sep<span class="op">=</span><span class="st">" "</span>, col.names<span class="op">=</span><span class="cn">FALSE</span>, row.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>ARNAQ identifies the target gene set by the directory name rather
than filename for brevity.</p>
</div>
<div class="section level3">
<h3 id="second-report">Second report<a class="anchor" aria-label="anchor" href="#second-report"></a>
</h3>
<p>With these support files, we can run a second, fuller report. This
time, we can specify that ARNAQ should create a folder <code>SVG</code>
in the working directory with <code>.svg</code> versions of all the
plots, for potential use in presentations or papers. You can inspect the
plots to check they match the ones in the report.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>svg.export<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; processing file: GSE155423_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: GSE155423_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_arnaq.html</span></span>
<span><span class="co">#&gt; processing file: GSE155423_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: GSE155423_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_arnaq.pdf</span></span></code></pre></div>
<p>You may observe that ARNAQ detects where it has already loaded a
version of a data file and uses that in preference to reading from disk.
If you need to start completely from scratch, you can use
<code><a href="../reference/arnaq_clear.html">arnaq_clear()</a></code>.</p>
<p>The new report will over-write the previous one. To preserve
different versions, we can use the <code>model.name</code> parameter, as
will be demonstrated next.</p>
<p>The output of this ARNAQ run, and subsequent variations, can be
viewed as vignettes in this package without needing to run the examples
in this code.</p>
</div>
</div>
<div class="section level2">
<h2 id="alternative-models">Alternative models<a class="anchor" aria-label="anchor" href="#alternative-models"></a>
</h2>
<p>Much of the QC that analysts perform on an RNASeq data set is about
inclusion and exclusion criteria for different samples. Apart from
idebntifying outliers, they may also be called upon to see how well
experimental groups segregate where whole groups are either included of
excluded, or to try to determine by how much borderline acceptable
samples might be influencing the clustering. Additionally, the project
might require the analyst to explore different comparisons within the
project completely separately.</p>
<p>For this reason, it is useful to consider different sets of samples
in this QC non-destructively.</p>
<p>This is done by means of the <code>sample.mask</code> parameter to
<code><a href="../reference/arnaq.html">arnaq()</a></code>. This parameter is a vector of
<code>TRUE</code>/<code>FALSE</code> values that signals whether a
sample should be included in the model or not. This approach eliminated
the need to keep multiple, slightly different copies of the data, which
may help when dealing with larger data sets.</p>
<p>We can create it using a simple function:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outlier.mask</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_outlier_mask.html">make_outlier_mask</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"D10PBS_1"</span>, <span class="st">"D15Myo_3"</span><span class="op">)</span><span class="op">)</span> <span class="co"># name the samples to exclude</span></span></code></pre></div>
<p>or by using group information from the <code>sample.metadata</code>
object created by our earlier runs:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">D10_D15.mask</span> <span class="op">&lt;-</span> <span class="va">sample.metadata</span><span class="op">$</span><span class="va">Day</span> <span class="op">!=</span> <span class="st">"D21"</span> <span class="co"># exclude 1 of the 3 groups for 'Day'</span></span></code></pre></div>
<p>This format makes it easy to combine and/or manipulate the masks:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">D10_D15.outlier.mask</span> <span class="op">&lt;-</span> <span class="va">outlier.mask</span> <span class="op">&amp;</span> <span class="va">D10_D15.mask</span></span></code></pre></div>
<p>Creating a report based on this mask is simply:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>model.name<span class="op">=</span><span class="st">"D10_D15_wo_outliers"</span>, sample.mask<span class="op">=</span><span class="va">D10_D15.outlier.mask</span><span class="op">)</span></span>
<span><span class="co">#&gt; factor levels were dropped which had no samples</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; processing file: GSE155423-D10-D15-wo-outliers_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-D10-D15-wo-outliers_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_D10_D15_wo_outliers_arnaq.html</span></span>
<span><span class="co">#&gt; processing file: GSE155423-D10-D15-wo-outliers_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-D10-D15-wo-outliers_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_D10_D15_wo_outliers_arnaq.pdf</span></span></code></pre></div>
<p>This will create a new report, and a new set of tables, in the
<code>QC</code> directory, based on only the samples retained by the
mask.</p>
<div class="section level3">
<h3 id="treat-groups-and-the-deseq2-model">
<code>treat.groups</code> and the DESeq2 model<a class="anchor" aria-label="anchor" href="#treat-groups-and-the-deseq2-model"></a>
</h3>
<p>The <code>DESeq2</code> model constructed during VSN normalisation
needs a model of full rank. When working with alternatie sets of
samples, it’s possible to end up with a subset that does not meet this
requirement. In this case, the <code>treat.groups</code> parameter can
be used to say which columns should be included.</p>
<p>By default, these will also be the columns used for the PCA plots.
The next section shows how you can override this behaviour.</p>
</div>
</div>
<div class="section level2">
<h2 id="output-options">Output options<a class="anchor" aria-label="anchor" href="#output-options"></a>
</h2>
<p>There are other aspects of ARNAQ output that you can alter.</p>
<div class="section level3">
<h3 id="pca-adjustments">PCA adjustments<a class="anchor" aria-label="anchor" href="#pca-adjustments"></a>
</h3>
<p>By default ARNAQ produce PCA plots for every categorical group, which
can be set up how you wish. However, you may wish to alter which columns
of <code>sample.metadata</code> are used for these plots. The depth of
the PCA plot will, by default include all components that account for at
least 10% of the total variability. This can also be adjusted.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>model.name<span class="op">=</span><span class="st">"alt_PCA"</span>, pca.groups<span class="op">=</span><span class="st">"Day"</span>, pca.depth<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; processing file: GSE155423-alt-PCA_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-alt-PCA_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_alt_PCA_arnaq.html</span></span>
<span><span class="co">#&gt; processing file: GSE155423-alt-PCA_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-alt-PCA_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_alt_PCA_arnaq.pdf</span></span></code></pre></div>
<p>If you are using <code>treat.groups</code> to restrict the DESeq2
model to only some metadata columns, you can still specify additional
columns for the PCA plots using <code>pca.groups</code>.</p>
</div>
<div class="section level3">
<h3 id="normalisation">Normalisation<a class="anchor" aria-label="anchor" href="#normalisation"></a>
</h3>
<p>The <code>normalise</code> parameter can be used to change how the
data is normalised. The default (<code>vsn</code>) uses a
variance-stabilising normalisation and will produce an accompanying plot
so the effectiveness of the normalisation can be assessed. The
alternative, <code>linear</code>, performs a simple linear
normalisation. Any other value will suppress normalisation and the raw
counts will be used.</p>
<p><strong>The VSN normalisation is impractically slow for large sets;
<code>linear</code> is strongly recommended for datasets with more than
30 samples.</strong></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>model.name<span class="op">=</span><span class="st">"linear"</span>, normalise<span class="op">=</span><span class="st">"linear"</span><span class="op">)</span></span>
<span><span class="co">#&gt; processing file: GSE155423-linear_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-linear_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_linear_arnaq.html</span></span>
<span><span class="co">#&gt; processing file: GSE155423-linear_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-linear_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_linear_arnaq.pdf</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="additional-optional-plots">Additional optional plots<a class="anchor" aria-label="anchor" href="#additional-optional-plots"></a>
</h2>
<div class="section level3">
<h3 id="sample-to-sample-scatterplots">Sample-to-sample scatterplots<a class="anchor" aria-label="anchor" href="#sample-to-sample-scatterplots"></a>
</h3>
<p>In some circumstances, you may wish to view scatterplots of one
sample against another; either because they are technical replicates or
you wish to check that the expression levels are appropriately
concordant between them.</p>
<p>You can specify which sample pairs using the
<code>scatter.pairs</code> parameter.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>model.name<span class="op">=</span><span class="st">"scatter"</span>, scatter.pairs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"D10Myo_1"</span>, <span class="st">"D10Myo_2"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"D15Myo_1"</span>, <span class="st">"D15Myo_2"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; processing file: GSE155423-scatter_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-scatter_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_scatter_arnaq.html</span></span>
<span><span class="co">#&gt; processing file: GSE155423-scatter_arnaq_tmp.rmd</span></span>
<span><span class="co">#&gt; output file: /home/runner/work/arnaq/arnaq/vignettes/GSE155423-scatter_arnaq_tmp.knit.md</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output created: QC/GSE155423_scatter_arnaq.pdf</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ercc-plots">ERCC plots<a class="anchor" aria-label="anchor" href="#ercc-plots"></a>
</h3>
<p>TODO</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by .</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
