<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Objects Created By ARNAQ • arnaq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Objects Created By ARNAQ">
<meta property="og:description" content="arnaq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">arnaq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/all-plots.html">Structure of the all_plots list</a>
    </li>
    <li>
      <a href="../articles/creating-example-reports.html">Creating Example Reports</a>
    </li>
    <li>
      <a href="../articles/files-created.html">Files Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/objects-created.html">Objects Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/workflow.html">Typical ARNAQ Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Objects Created By ARNAQ</h1>
            
      
      
      <div class="hidden name"><code>objects-created.Rmd</code></div>

    </div>

    
    
<p>ARNAQ creates a number of R objects in the course of a run, which
remain available after the run is completed. The rationale for this is
twofold. Firstly, some of this data (count tables read from files, for
example) is useful for any downstream analysis and it is useful to
already have this available in an R session. Secondly, the user can
alter the appearance of the final report by altering plots and/or
changing which sections are included. The default versions will probably
be suitable in most instances, but it is impractical to anticipate every
change a user might need to make to the final appearance of the plot or
report and this method provides an accessible method of customisation
without overencumbering the main functions with options.</p>
<p>All of these objects will be overwritten by a subsequent ARNAQ run in
the same session. As the intention is for sequential runs for the same
project to be iterations towards a final version of the report, this
should not be a problem. If necessary, the R session object can be
copied into a different directory or the objects renamed</p>
<p>These objects can broken down into three categories:</p>
<div class="section level2">
<h2 id="report-preparation">Report Preparation<a class="anchor" aria-label="anchor" href="#report-preparation"></a>
</h2>
<p>These objects are used internally by ARNAQ when preparing a report,
but also need to be supplied when recreating a report after fine-tuning
plots. <code><a href="../reference/arnaq.create.report.html">arnaq.create.report()</a></code> uses default values for some
of its parameters that match these object names.</p>
<div class="section level3">
<h3 id="all_plots">all_plots<a class="anchor" aria-label="anchor" href="#all_plots"></a>
</h3>
<p>This named list contains all the plot objects that are used in the
report. Full details are provided in a <a href="all-plots.html">separate
vignette</a>.</p>
</div>
<div class="section level3">
<h3 id="arnaq-plot-flags">arnaq.plot.flags<a class="anchor" aria-label="anchor" href="#arnaq-plot-flags"></a>
</h3>
<p>A named list of logical values that controls which sections and plots
are included when<br><code><a href="../reference/arnaq.create.report.html">arnaq.create.report()</a></code> is used. This object is created with
values that reflect what data was available for the run. Specific plots
can be set to <code>FALSE</code> to exclude them anyway. Changing a
value to <code>TRUE</code> is not recommended as it is unlikely that the
underlying plot the report would need is available.</p>
</div>
<div class="section level3">
<h3 id="arnaq-report-template">arnaq.report.template<a class="anchor" aria-label="anchor" href="#arnaq-report-template"></a>
</h3>
<p>This is the file path to the ARNAQ report template, as given in
<code>resources.yml</code>.</p>
</div>
<div class="section level3">
<h3 id="arnaq-run">arnaq.run<a class="anchor" aria-label="anchor" href="#arnaq-run"></a>
</h3>
<p>A list containing supplementary data about the ARNAQ run, covered
later.</p>
</div>
<div class="section level3">
<h3 id="qc-name">qc.name<a class="anchor" aria-label="anchor" href="#qc-name"></a>
</h3>
<p>The extended name for this ARNAQ run, based on the project id given
in <code>resources.yml</code> and the <code>model.name</code> paramter
(if any) given to <code><a href="../reference/arnaq.html">arnaq()</a></code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="important-data">Important Data<a class="anchor" aria-label="anchor" href="#important-data"></a>
</h2>
<p>These are the primary pieces of data associated with the project;
those most likely to be useful in downstream analysis or to be inspected
within R.</p>
<div class="section level3">
<h3 id="count-data">count.data<a class="anchor" aria-label="anchor" href="#count-data"></a>
</h3>
<p>The complete raw counts table, as read in from the data file and
without any filtering.</p>
</div>
<div class="section level3">
<h3 id="gene-masks">gene.masks<a class="anchor" aria-label="anchor" href="#gene-masks"></a>
</h3>
<p>This is a list of logical masks for genes, rather than samples. Most
categories are pulled from the <code>genome_reference</code> file
specified in <code>resources.yml</code>.</p>
<ul>
<li>
<code>Genes</code> denotes identifiers listed as genes.</li>
<li>
<code>ERCC</code> are for ERCC spike-ins. Any gene identifier that
starts with ERCC will be considered as such, if ERCC functionality is
turned on in <code><a href="../reference/arnaq.html">arnaq()</a></code>.</li>
<li>
<code>protein_coding</code>, <code>miRNA</code>,
<code>lncRNA</code>, <code>sRNA</code>, <code>snoRNA</code>,
<code>snRNA</code> and <code>scaRNA</code>, as defined in the .gtf file,
are present. tRNA is listed as ‘sRNA’ in the Ensembl data file.</li>
<li>
<code>ncRNA</code> represents the union of the previous set of
categories.</li>
</ul>
<p>For all masks except ERCC, the main masks excludes genes with no
reads for any sample. If you want an equivalent mask that includes these
zero read genes, append <code>_Z</code> to the name.</p>
<p>The ERCC mask is not filtered in this way, on the basis that missing
spike-ins are anticipated to be of interest when checking observed
against expected reads.</p>
</div>
<div class="section level3">
<h3 id="norm-data">norm.data<a class="anchor" aria-label="anchor" href="#norm-data"></a>
</h3>
<p>The normalised count data. This is filtered by both
<code>sample.mask</code> and <code>gene.masks$Genes</code>. The type of
normalisation performed in set as a paramter in <code><a href="../reference/arnaq.html">arnaq()</a></code>. In
the special case where no normalisation is performed, this will be the
filtered raw count matrix.</p>
</div>
<div class="section level3">
<h3 id="sample-mask">sample.mask<a class="anchor" aria-label="anchor" href="#sample-mask"></a>
</h3>
<p>The sample mask supplied to <code><a href="../reference/arnaq.html">arnaq()</a></code> is stored here. The
<code>count.data</code> and <code>sample.metadata</code> entries below
are <strong>not</strong> filtered by this mask.</p>
</div>
<div class="section level3">
<h3 id="sample-metadata">sample.metadata<a class="anchor" aria-label="anchor" href="#sample-metadata"></a>
</h3>
<p>The complete sample metadata table, as read in from the data file and
without any filtering. You can use the group columns of this object to
create sample masks for use with <code><a href="../reference/arnaq.html">arnaq()</a></code>, instead of or in
combination with <code><a href="../reference/make_outlier_mask.html">make_outlier_mask()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="species-gtf">species.gtf<a class="anchor" aria-label="anchor" href="#species-gtf"></a>
</h3>
<p>The Ensembl .gtf file used to find gene categories is stored in the
session.</p>
</div>
</div>
<div class="section level2">
<h2 id="other-data-and-metadata">Other Data and Metadata<a class="anchor" aria-label="anchor" href="#other-data-and-metadata"></a>
</h2>
<p>Other data is placed in the <code>arnaq.run</code> list object under
the following names, to avoid cluttering up the main session. This
includes complete R copies of some of the source files for the ARNAQ
run.</p>
<p>These entries are separated by category here for convenience.</p>
<div class="section level3">
<h3 id="run-information">Run Information<a class="anchor" aria-label="anchor" href="#run-information"></a>
</h3>
<p>Information about the run itself.</p>
<div class="section level4">
<h4 id="version">version<a class="anchor" aria-label="anchor" href="#version"></a>
</h4>
<p>The version of ARNAQ used.</p>
</div>
<div class="section level4">
<h4 id="date">date<a class="anchor" aria-label="anchor" href="#date"></a>
</h4>
<p>The date the run was started.</p>
</div>
<div class="section level4">
<h4 id="resources">resources<a class="anchor" aria-label="anchor" href="#resources"></a>
</h4>
<p>The file path of <code>resources.yml</code>.</p>
</div>
<div class="section level4">
<h4 id="biotype-conversion-table">biotype.conversion.table<a class="anchor" aria-label="anchor" href="#biotype-conversion-table"></a>
</h4>
<p>The table used to map the full set of Ensembl biotypes to more
readable consolidated categories.</p>
</div>
<div class="section level4">
<h4 id="project-id">project.id<a class="anchor" aria-label="anchor" href="#project-id"></a>
</h4>
<p>The project name given in <code>resources.yml</code>. Used in
combination with the <code>model.name</code> parameter of
<code><a href="../reference/arnaq.html">arnaq()</a></code> to give the full name of a run.</p>
</div>
<div class="section level4">
<h4 id="genome-file">genome.file<a class="anchor" aria-label="anchor" href="#genome-file"></a>
</h4>
<p>The .gtf file used to determine gene biotype, if any was
provided.</p>
</div>
<div class="section level4">
<h4 id="out-directory">out.directory<a class="anchor" aria-label="anchor" href="#out-directory"></a>
</h4>
<p>The directory the output files appeared in. Currently
<code>./QC/</code>.</p>
</div>
<div class="section level4">
<h4 id="model-formula">model.formula<a class="anchor" aria-label="anchor" href="#model-formula"></a>
</h4>
<p>The model formula, based on the treatment groups specified, as used
by DESeq2 for normalisation.</p>
</div>
</div>
<div class="section level3">
<h3 id="input-data">Input Data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>Other data as read from external files.</p>
<div class="section level4">
<h4 id="read-summary">read.summary<a class="anchor" aria-label="anchor" href="#read-summary"></a>
</h4>
<p>The table of mapped reads assigned to categories, as per
featureCounts or equivalent, as in the <code>summary_table</code> entry
of <code>resources.yml</code>, if available.</p>
</div>
<div class="section level4">
<h4 id="count-metrics">count.metrics<a class="anchor" aria-label="anchor" href="#count-metrics"></a>
</h4>
<p>The table of metrics for samples, as calculated by Picard or
equivalent, as in the <code>metrics_table</code> and
<code>duplication_table</code> of <code>resources.yml</code>, if
availble.</p>
</div>
<div class="section level4">
<h4 id="ercc-data">ERCC.data<a class="anchor" aria-label="anchor" href="#ercc-data"></a>
</h4>
<p>The table of counts of ERCC spike-ins per samples, if the
<code>ERCC</code> paramter is <code>TRUE</code> in the
<code><a href="../reference/arnaq.html">arnaq()</a></code> run.</p>
</div>
</div>
<div class="section level3">
<h3 id="derived-data">Derived Data<a class="anchor" aria-label="anchor" href="#derived-data"></a>
</h3>
<p>Data that is calculated from the sets loaded in.</p>
<div class="section level4">
<h4 id="complexity-matrix">complexity.matrix<a class="anchor" aria-label="anchor" href="#complexity-matrix"></a>
</h4>
<p>The complexity matrix is a long format dataframe based on sample and
top N genes, with a value of the proportion of the sample reads
contained within that top N.</p>
</div>
<div class="section level4">
<h4 id="cpm-matrix">cpm.matrix<a class="anchor" aria-label="anchor" href="#cpm-matrix"></a>
</h4>
<p>The cpm matrix as calculated based on the normalised read counts.</p>
</div>
<div class="section level4">
<h4 id="norm-pca">norm.PCA<a class="anchor" aria-label="anchor" href="#norm-pca"></a>
</h4>
<p>The Principle Components Analysis based on the normalised read
counts. This is a list with three members. <code>PC</code> is the matrix
itself. <code>Vars</code> is the vector giving the propotions of
variability explained by each component. <code>CumulVars</code> is the
cumulative version of the previous.</p>
</div>
<div class="section level4">
<h4 id="ercc-data-cpm">ERCC.data.cpm<a class="anchor" aria-label="anchor" href="#ercc-data-cpm"></a>
</h4>
<p>The cpm version of <code>ERCC.data</code>.</p>
</div>
<div class="section level4">
<h4 id="biotype-counts">biotype.counts<a class="anchor" aria-label="anchor" href="#biotype-counts"></a>
</h4>
<p>Table of reads collated into the Ensembl biotype categories, in the
simplified mapping provided by the support file
<code>biotype_conversion</code> in <code>resources.yml</code>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by .</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
