<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Typical ARNAQ Workflow • arnaq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Typical ARNAQ Workflow">
<meta property="og:description" content="arnaq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">arnaq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/all-plots.html">Structure of the all_plots list</a>
    </li>
    <li>
      <a href="../articles/creating-example-reports.html">Creating Example Reports</a>
    </li>
    <li>
      <a href="../articles/files-created.html">Files Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/objects-created.html">Objects Created By ARNAQ</a>
    </li>
    <li>
      <a href="../articles/workflow.html">Typical ARNAQ Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Typical ARNAQ Workflow</h1>
            
      
      
      <div class="hidden name"><code>workflow.Rmd</code></div>

    </div>

    
    
<p>ARNAQ is a tool for analysts working with RNA sequencing data,
intended to make the process of assessing a data set for QC issues
before downstream analysis quick and simple. It complements, rather than
replaces, QC metrics generated during sequencing, or by the processing
steps immediately following sequencing. ARNAQ is designed to be
experiment-aware, allowing users to identify issues that would not
necessarily be apparent without that context, such as:</p>
<ul>
<li>Sample swaps between groups</li>
<li>Batch effects</li>
<li>Systematic variation in read depth or library complexity between
experimental groups</li>
<li>Poor differentiation between different groups of samples</li>
</ul>
<p>It does so in a way that’s designed to minimise duplicative effort by
the analysis, provide justification for modelling decisions such as
outlier removal, and act as a record of exactly what steps were taken in
the preparation of a particular run. Analyst time is spent examining the
data.</p>
<p>The html report can be used by the analyst to review the data set,
but is also in a format that can be sent to collaborators unfamiliar
with R or the types of QC plot typically made. Optionally, high quality
<code>.svg</code> files can be exported for every plot included in the
report, ready for inclusion in papers.</p>
<p>Finally, an ARNAQ run creates objects in the R session for the
normalised data, sample metadata, and more; simplifying the process of
performing downstream analysis such as differential expression
analysis.</p>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>The full workflow for ARNAQ looks like this:</p>
<ul>
<li>Create data and metrics files</li>
<li>Create <code>resources.yml</code> and <code>samples.txt</code>
</li>
<li>Run ARNAQ</li>
<li>Iterate
<ul>
<li>Investigate possible QC issues</li>
<li>Rerun ARNAQ with different parameters</li>
</ul>
</li>
<li>(Optionally) Fine tune plots
<ul>
<li>Recreate report</li>
</ul>
</li>
<li>(Optionally) Export <code>.svg</code> versions of the plots</li>
<li>Downstream analysis</li>
</ul>
<p>ARNAQ will automatically log its on-screen output to a file each time
it is run, but it is recommended that you also keep a separate record of
what R code you used when running ARNAQ.</p>
<div class="section level3">
<h3 id="create-data-and-metrics-files">Create data and metrics files<a class="anchor" aria-label="anchor" href="#create-data-and-metrics-files"></a>
</h3>
<p>Create your count tables using your preferred tools. ARNAQ expects
the standard format of text files with a header row containing sample
names, with one column for Ensembl gene identifiers and subsequent
columns for sample read counts.</p>
<p>Optionally, summaries of read assignment can be attached (as from
featureCounts)</p>
<p>Also optionally, consolidated metrics files from Picard (RNAMetrics
and MarkDuplicates) can also be provided.</p>
</div>
<div class="section level3">
<h3 id="create-resources-yml-and-samples-txt">Create <code>resources.yml</code> and <code>samples.txt</code><a class="anchor" aria-label="anchor" href="#create-resources-yml-and-samples-txt"></a>
</h3>
<p>These two files contain information about the data that is not
anticipated to change for different ARNAQ runs.
<code>resources.yml</code> contains non-sample-specific metadata and
<code>samples.txt</code> contains sample metadata. These are used to
establish which genome to use, define experimental groups, and point to
the file locations of the raw count data.</p>
<p>By making these files rather than paramters within R, it is easier to
automate their generation as part of your pipeline.</p>
</div>
<div class="section level3">
<h3 id="run-arnaq">Run ARNAQ<a class="anchor" aria-label="anchor" href="#run-arnaq"></a>
</h3>
<p>If the above files are prepared, the initial run of ARNAQ can be as
simple as:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>This will run the tool with default parameters. Settings for which
organism, where the files are located, etc., are handled by
<code>resources.yml</code>. The parameters to <code><a href="../reference/arnaq.html">arnaq()</a></code> allow
you greater control over the process.</p>
</div>
<div class="section level3">
<h3 id="iterate">Iterate<a class="anchor" aria-label="anchor" href="#iterate"></a>
</h3>
<p>The report may indicate potential QC issues with the data set, such
as batch effects and failed samples. In some cases, these will have been
identified by earlier QC and ARNAQ helps confirm them. In particular,
where a sample is borderline the plots in ARNAQ give some information as
to <em>how much</em> the issue might be affecting the data for that
sample.</p>
<p>It is common to need to iteratively run reports until you are
satisfied no more outliers remain. In particular, bad samples can skew
some of the diagnostic plots and make it harder to determine if the
experimental groups are egregated in the plots, or if there are any less
dramatic outliers waiting to be identified.</p>
<p>Once you are satisfied that no more issues remain, the most recent
version of the report can be used as the final one.</p>
<p>Suggestions on how to handle two of the most common issues are
below.</p>
<div class="section level4">
<h4 id="outliers">Outliers<a class="anchor" aria-label="anchor" href="#outliers"></a>
</h4>
<p>The most common parameter change is to exclude one or more samples
from the run. This can be done using
<code>make_sample_mask()</code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outliers</span> <span class="op">&lt;-</span> <span class="fu">make_sample_mask</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BadSampleA"</span>, <span class="st">"BadSampleB"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>model.name <span class="op">=</span> <span class="st">"outliers_removed"</span>, sample.mask <span class="op">=</span> <span class="va">outliers</span><span class="op">)</span></span></code></pre></div>
<p><em>Note that as <code>make_sample_mask()</code> uses the sample
metadata object created by an ARNAQ run, you need to perform an initial
run before using it.</em></p>
<p>As <code>make_sample_mask()</code> returns a
<code>TRUE</code>/<code>FALSE</code> vector, it is easy to make
compatible masks using logical operators and the sample metadata:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group.AB.mask</span> <span class="op">&lt;-</span> <span class="va">sample.metadata</span><span class="op">$</span><span class="va">Group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/arnaq.html">arnaq</a></span><span class="op">(</span>model.name <span class="op">=</span> <span class="st">"groups_A_and_B"</span>, sample.mask <span class="op">=</span> <span class="va">group.AB.mask</span><span class="op">)</span> <span class="co"># Will only consider samples in groups A and B</span></span></code></pre></div>
<p>In both examples, a <code>model.name</code> parameter is given to
differentiate the output. ARNAQ will overwrite the objects in the
session with new versions each time it is run, but the output files will
have the model name included in them, keeping multiple versions die by
side.</p>
</div>
<div class="section level4">
<h4 id="batch-effects">Batch effects<a class="anchor" aria-label="anchor" href="#batch-effects"></a>
</h4>
<p>Although the PCA plot for a project may be very persuasive with
regard to a potential batch effect, it cannot be taken as such without
corroboration external to the read count data. If such corroboration
exists, then it may be convenient to include batch information as an
extra column in <code>samples.txt</code>.</p>
<p>Not only will this include batch as a group for PCA plots in the
report, but it will make sure that the batch effect is in the
<code>sample.metadata</code> object ready for any downstream
analysis.</p>
</div>
</div>
<div class="section level3">
<h3 id="fine-tune-plots">Fine tune plots<a class="anchor" aria-label="anchor" href="#fine-tune-plots"></a>
</h3>
<p><em>Optional.</em></p>
<p>You may wish to alter individual plots to better display points of
interest. Care has been taken to make the default plots as robust as
possible, whatever the data, but no uncurated method will ever be
perfect. All of the plots that appear in the report are stored in the
<code>all_plots</code> list, and as <code>ggplot2</code> objects you can
adjust their presentation as you wish. Details on the structure of
<code>all_plots</code> is given <a href="all-plots.html">here</a>.</p>
<p>To produce the most recent report with these new plots, you use
<code><a href="../reference/arnaq.create.report.html">arnaq.create.report()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="export-to--svg">Export to <code>.svg</code><a class="anchor" aria-label="anchor" href="#export-to--svg"></a>
</h3>
<p><em>Optional.</em></p>
<p>The <code>export.svg</code> parameter to <code><a href="../reference/arnaq.html">arnaq()</a></code> will
create SVG format exports of every plot in the <code>QC/svg</code>
folder.</p>
<p>If you wish to export them without running the entire tool, the
<code><a href="../reference/export_all_svg.html">export_all_svg()</a></code> function will accomplish this.</p>
</div>
<div class="section level3">
<h3 id="downstream-analysis">Downstream analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h3>
<p>The objects <code>sample.metadata</code>, <code>count.data</code> and
<code>norm.data</code> in session can be used directly with differential
expression packages such as <code>DESeq2</code>, <code>edgeR</code> or
your preferred tool. Alternatively, the data files created by ARNAQ can
be used.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by .</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
